clib/cThread.ml

```diff
@@ -107,3 +107,18 @@ let mask_sigalrm f x =
 
 let create f x =
   Thread.create (mask_sigalrm f) x
+
+(*
+  Atomic mutex lock taken from https://gitlab.com/gadmm/memprof-limits/-/blob/master/src/thread_map.ml#L23-34
+  Critical sections :
+   - Mutex.lock does not poll on leaving the blocking section
+     since 4.12.
+   - Never inline, to avoid theoretically-possible reorderings with
+     flambda.
+     (workaround to the lack of masking)
+*)
+let[@inline never] with_lock m ~scope =
+  let () = Mutex.lock m (* BEGIN ATOMIC *) in
+  match (* END ATOMIC *) scope () with
```

Comment:
```suggestion
  let () = Mutex.lock m (* BEGIN ATOMIC : exit from C code of lock *) in
  match (* END ATOMIC : exit from C code setting exception handler*)
    scope ()
  with
```
(assuming I got it right)

Reply:
I am not sure to understand these precisions or how precise you want it. The beginning of the atomic operation is the acquisition of the (system) resource (here when the posix lock call returns in the C function). The end of the atomic operation is the initialisation of the (semantic) resource (so here setting up the exception handler).